pax/plugins/peak_processing/PeakAreaCorrections.py

```diff
@@ -64,3 +69,223 @@ def transform_event(self, event):
+                if(area_nan_saturate>1e3):
+                    peak.area_midpoint, peak.range_area_decile = compute_area_deciles(wf_sum_nan_saturate)
+                    peak.range_area_decile*=dt
+                    #print('width after correction: ',peak.range_area_decile)
+                
+        return event
+    
+    
+    
+def compute_area_deciles(w):
+    """Return (index of mid area, array of the 0th ... 10 th area decile ranges in samples) of w
+    e.g. range_area_decile[5] = range of 50% area = distance (in samples)
+    between point of 25% area and 75% area (with boundary samples added fractionally).
+    First element (0) of array is always zero, last element (10) is the length of w in samples.
+    """
+    fractions_desired = np.linspace(0, 1, 21)
+    index_of_area_fraction = np.ones(len(fractions_desired)) * float('nan')
+    integrate_until_fraction(w, fractions_desired, index_of_area_fraction)
+    return index_of_area_fraction[10], (index_of_area_fraction[10:] - index_of_area_fraction[10::-1]),
+
+
+# @numba.jit(numba.void(numba.float32[:], numba.float64[:], numba.float64[:]),
+#            nopython=True, cache=True)
+# For some reason numba doesn't clean up its memory properly for this function... leave it in python for now
+def integrate_until_fraction(w, fractions_desired, results):
+    """For array of fractions_desired, integrate w until fraction of area is reached, place sample index in results
+    Will add last sample needed fractionally.
+    eg. if you want 25% and a sample takes you from 20% to 30%, 0.5 will be added.
+    Assumes fractions_desired is sorted and all in [0, 1]!
+    """
+    area_tot = w.sum()
+    fraction_seen = 0
+    current_fraction_index = 0
+    needed_fraction = fractions_desired[current_fraction_index]
+    for i, x in enumerate(w):
+        # How much of the area is in this sample?
+        fraction_this_sample = x/area_tot
+        # Will this take us over the fraction we seek?
+        # Must be while, not if, since we can pass several fractions_desired in one sample
+        while fraction_seen + fraction_this_sample >= needed_fraction:
+            # Yes, so we need to add the next sample fractionally
+            area_needed = area_tot * (needed_fraction - fraction_seen)
+            results[current_fraction_index] = i + area_needed/x
+            # Advance to the next fraction
+            current_fraction_index += 1
+            if current_fraction_index > len(fractions_desired) - 1:
+                return
+            needed_fraction = fractions_desired[current_fraction_index]
+        # Add this sample's area to the area seen, advance to the next sample
+        fraction_seen += fraction_this_sample
+    if needed_fraction == 1:
+        results[current_fraction_index] = len(w)
+    else:
+        # Sorry, can't add the last fraction to the error message: numba doesn't allow it
+        raise RuntimeError("Fraction not reached in waveform? What the ...?")
+
+
```

Comment:
__[flake8]__
* [W391] blank line at end of file

<!-- Comment by [SideCI](https://sideci.com) -->